doc/ensemble.rst

```diff
@@ -0,0 +1,1190 @@
+
+1. :ref:`Application Producer Script<app_prod_prefix_ensemble>`: This script is encapsulated
+   in a SmartSim ``Ensemble`` within the ``Experiment`` driver script. Prefixing is enabled
+   on the ``Ensemble``. The producer script puts NumPy tensors on an ``Orchestrator``
+   launched in the ``Experiment`` driver script. The ``Ensemble`` creates two
+   identical ``Ensemble`` members. The producer script is executed
+   in both ``Ensemble`` members to send two prefixed tensors to the ``Orchestrator``.
+   The source code example is available in the dropdown below for convenient customization.
+
+.. dropdown:: Application Producer Script source code
+
+    .. literalinclude:: ../tutorials/doc_examples/ensemble_doc_examples/application_producer_script.py
+
+1. :ref:`Application Consumer Script<app_con_prefix_ensemble>`: This script is encapsulated
+   within a SmartSim ``Model`` in the ``Experiment`` driver script. The script requests the
+   prefixed tensors placed by the producer script. The source code example is available in
+   the dropdown below for convenient customization.
+
+.. dropdown:: Application Consumer Script source code
+
+    .. literalinclude:: ../tutorials/doc_examples/ensemble_doc_examples/application_consumer_script.py
+
+1. :ref:`Experiment Driver Script<exp_prefix_ensemble>`: The driver script launches the
+   ``Orchestrator``, the ``Ensemble`` (which sends prefixed keys to the ``Orchestrator``),
+   and the ``Model`` (which requests prefixed keys from the ``Orchestrator``). The
+   ``Experiment`` driver script is the centralized spot that controls the workflow.
+   The source code example is available in the dropdown below for convenient execution and
+   customization.
+
+.. dropdown:: Experiment Driver Script source code
+
+    .. literalinclude:: ../tutorials/doc_examples/ensemble_doc_examples/experiment_driver.py
+
+.. _app_prod_prefix_ensemble:
+-------------------------------
+The Application Producer Script
+-------------------------------
+In the ``Experiment`` driver script, we instruct SmartSim to create an ``Ensemble`` comprised of
+two duplicate members that execute this producer script. In the producer script, a SmartRedis ``Client`` sends a
+tensor to the ``Orchestrator``. Since the ``Ensemble`` members are identical and therefore use the same
+application code, two tensors are sent to the ``Orchestrator``. Without prefixing enabled on the ``Ensemble``
+the keys can be overwritten. To prevent this, we enable key prefixing on the ``Ensemble`` in the driver script
+via ``Ensemble.enable_key_prefixing()``. When the producer script is executed by each ``Ensemble`` member, a
+tensor is sent to the ``Orchestrator`` with the ``Ensemble`` member `name` prepended to the tensor `name`.
+
+Here we provide the producer script that is applied to the ``Ensemble`` members:
+
+.. literalinclude:: ../tutorials/doc_examples/ensemble_doc_examples/application_producer_script.py
+  :language: python
+  :linenos:
```

Comment:
Maybe do this as a dropdown like in the other example?

Reply:
the dropdowns are in the examples summary! At the top of the example, let me know if you think moving them down would be better